text/0750-feature-flag-advisor.md

```diff
@@ -0,0 +1,398 @@
+    Feature Flag                              Recommended Value            User Value
+  * @aws-cdk/...                              true                         - <unset>
+  * @aws-cdk/...                              true                         - <unset>
+  Here is the difference:
+    Resources
+    [~] AWS::S3::Bucket MyBucket
+    └─ [~] Properties
+        └─ [~] Encryption
+            ├─ [-] None
+            └─ [+] ServerSideEncryptionConfiguration:
+                    - ...
+ 
+    [~] AWS::CloudFront::Distribution MyDistribution
+    └─ [~] Properties
+        └─ [~] DefaultSecurityPolicy
+            ├─ [-] TLSv1.0
+            └─ [+] TLSv1.2_2021
+                    - ...
+    Number of stacks with differences: 2
+  Would you like to accept or reject these changes? (Accept/reject)
+> Accept
+  Resynthesizing...
+```
+
+The third option, `set unconfigured flags to their implied configuration`, is equivalent to the command `cdk flags
+--set --default --unconfigured`. If you want to ensure the unconfigured flags do not interfere with your application,
+this option changes the unconfigured feature flags to its default values. For example, if
+`@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021` is unconfigured, it leads to the notification
+appearing after running `cdk synth`. However, if you set the flag to its default state (false), it will be
+configured, turned off, and have no impact on your application whatsoever.
+
+```
+> cdk flags --set --default --unconfigured
+    Feature Flag                              Recommended Value            User Value
+  * @aws-cdk/...                              true                         - <unset>
+  * @aws-cdk/...                              true                         - <unset>
+  Here is the difference:
+  
+  Would you like to accept or reject these changes? (Accept/reject)
+> Accept
+  Resynthesizing...
+```
+
+The fourth option, `modify a specific flag`, asks you which flag you want to modify. You are prompted to enter
+the name of the flag and choose an existing state after. Use this command if there is a specific bug fix you
+would like to incorporate into your application.
+
+```
+  Which flag would you like to modify?
+> "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021"
+  What value would you like to change "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021" to?
+    > **true**
+      false
+  Here is the difference:
+    Resources
+    [~] AWS::CloudFront::Distribution MyDistribution
+    └─ [~] Properties
+        └─ [~] DefaultSecurityPolicy
+            ├─ [-] TLSv1.0
+            └─ [+] TLSv1.2_2021
+                    - ...
+    Number of stacks with differences: 1
+  Would you like to accept or reject these changes? (Accept/reject)
+> Accept
+  Resynthesizing...
+```
+
+To achieve the same result, you can also run `cdk flags --set "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021" --value="true"`.
```

Comment:
If I set a flag with a non-recommended value this way, is `cdk synth` still going to notify that I have a non-recommended value?